test/test-parse-typescript-class.cpp

```diff
@@ -1933,6 +1933,26 @@ TEST_F(test_parse_typescript_class,
         }));
   }
 }
+
+TEST_F(test_parse_typescript_class,
+       class_keyword_with_escape_sequence) {
+  test_parser p(
+      u8"class C {\n"_sv
+      u8"  \\u{63}onstructor() {}"_sv // equivalent to: constructor() {}
+      u8"}"_sv,
+      typescript_options, capture_diags);
+  p.parse_and_visit_statement();
+
+  EXPECT_THAT(
+      p.errors,
+      ElementsAreArray({
+          DIAG_TYPE_OFFSETS(
+              p.code,
+              diag_keyword_contains_escape_characters,  //
+              escape_character_in_keyword, strlen(u8"class C {\n  "), 
+              u8"\\u{63}onstructor"_sv),
+      }));
+}
```

Comment:
Please include a test for JavaScript.

Please include a test for the `\u{63}onstructo\u{72}` example.

Please include a test for `interface`.

Reply:
Added following tests,
1. A test to ensure no diag raised for JavaScript code where keyword contains escape.
2. A test for the \u{63}onstructo\u{72} example.
3. Two tests for interface,
        - a test for interface with escaped identifier
        - a test for interface with escaped keyword